samples/v1beta1/buildstrategy/multiarch-native-buildah/buildstrategy_multiarch_native_buildah_cr.yaml

```diff
@@ -0,0 +1,537 @@
+            elif [ "${status}" == "parse_build_contexts" ]; then
+              buildArgs+=("--build-context" "${arg}")
+            elif [ "${status}" == "parse_registries_block" ]; then
+              registriesBlock="${registriesBlock}'${arg}', "
+            elif [ "${status}" == "parse_registries_insecure" ]; then
+              registriesInsecure="${registriesInsecure}'${arg}', "
+            elif [ "${status}" == "parse_registries_search" ]; then
+              registriesSearch="${registriesSearch}'${arg}', "
+            else
+              echo "[ERROR] Invalid usage"
+              exit 1
+            fi
+          done
+
+          # Verify the existence of the context directory
+          if [ ! -d "${context}" ]; then
+            echo -e "[ERROR] The context directory '${context}' does not exist."
+            echo -n "ContextDirNotFound" > '$(results.shp-error-reason.path)'
+            echo -n "The context directory '${context}' does not exist." > '$(results.shp-error-message.path)'
+            exit 1
+          fi
+          cd "${context}"
+
+          # Verify the existence of the Dockerfile
+          if [ ! -f "${dockerfile}" ]; then
+            echo -e "[ERROR] The Dockerfile '${dockerfile}' does not exist."
+            echo -n "DockerfileNotFound" > '$(results.shp-error-reason.path)'
+            echo -n "The Dockerfile '${dockerfile}' does not exist." > '$(results.shp-error-message.path)'
+            exit 1
+          fi
+          echo "[INFO] Creating registries config file..."
+          touch /tmp/registries.conf
+          if [ "${registriesSearch}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.search]
+          registries = [${registriesSearch::-2}]
+          EOF
+          fi
+          if [ "${registriesInsecure}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.insecure]
+          registries = [${registriesInsecure::-2}]
+          EOF
+          fi
+          if [ "${registriesBlock}" != "" ]; then
+            cat <<EOF >>/tmp/registries.conf
+          [registries.block]
+          registries = [${registriesBlock::-2}]
+          EOF
+          fi
+
+          if [ "${#runtime_stage_from_image}" -gt 0 ]; then
+            echo "[INFO] Replacing the runtime stage image to extend from"
+            buildArgs+=("--build-context" "$(tac "${dockerfile}" | grep -m1 -i -E '^[ ]*FROM' | \
+              sed -n '0,/p/s/^[ ]*from[ ]\+\([^ ]*\)[ ]*\(as.*$\)\{0,1\}$/\1/Ip')=${runtime_stage_from_image}")
+          fi
+          # TODO: what image including kubectl can we use?
+          echo "[INFO] Downloading kubectl..."
+          kube_version=$(curl --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+            "https:${KUBERNETES_PORT#tcp:}/version" | sed -n 's/^.*gitVersion.*"v\(.*\)+.*$/\1/p')
+          arch=$(uname -m | sed 's/x86_64/amd64/;s/aarch64/arm64/')
+          curl -L -o /usr/local/bin/kubectl \
```

Comment:
just adding a note for this as this will be executed every time a new build is spawn and reaches an outside network. I'm not sure if we can have an image with kubectl (and how we should consider the right version to use).

Reply:
Per discussion we had live - agree that this step can be avoided if a single image with all needed binaries is published.